include/range/v3/view_interface.hpp

```diff
@@ -297,6 +297,19 @@ namespace ranges
             {
                 return Slice{}(detail::move(derived()), offs.from, offs.to);
             }
+            /// Returns a reference to the element at specified location pos, with bounds checking.
+            template<typename D = Derived,
+                CONCEPT_REQUIRES_(Same<D, Derived>() && RandomAccessRange<D>())>
+            auto at( range_difference_type_t<D> n) ->
+                decltype(std::declval<D &>().begin()[n])
+            {
+                using size_type = decltype( derived().size() );
```

Comment:
You haven't required `SizedRange<D>()`, so this may fail to instantiate. Once you have required `SizedRange`, you can simply use `range_size_type_t<Derived>` for its size type.